### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by max-veit**
This update unifies the Python and C++ documentation projects into one Sphinx project.

Also brought in some previously orphaned files, like the SOAP LaTeX derivation and the IPython notebook tutorials.

Marked as "help wanted" because:

- The developer-facing documentation is still woefully inadequate, and the C++ code in particular has a large proportion of undocumented and unexplained (or even inexplicable) code. All public-facing classes, functions, and members _need_ documentation, if only a sentence or two of plain English to say what it does. This is something we should be doing _as_ we develop, not after the fact. Over the next few weeks, I will be finding some of the worst examples and tracking down their perpetrators. I have a `git blame` and I'm not afraid to use it. &lt;/rant&gt;
- The Sphinx documentation generated by `breathe` is good, but not perfect -- in particular, it requires a lot of organizing by hand. To this end, the Doxygen HTML docs have been left as a reference, but we eventually want to replace them with the sphinx/breathe equivalents.
- Sphinx still spits out loads of warnings when compiling (especially from `breathe`) and I still can't figure out whether they're benign or a sign of more serious problems.
- Python autodoc still needs to be added
- Renamed dev_doc target to doc, please use now **make doc** to make the documentation, and install pandoc on your system and the python library, if you have not yet

I hope with this update we can start incorporating (at least minimal, developer-facing) documentation into our workflow, since it's now quite easy to build the whole documentation and check that everything looks right. (and yes, I will be looking for this in future pull requests.)

Thanks to Klim for organizing and revising the docs, and working on the tutorials.

### CODE REVIEW

**Comment by max-veit** [comment]
Oh yeah, and the installation instructions (currently a standalone `.rst`) should really be coupled to the project README.md file. I think there's an easy way to do this IIRC

**Comment by mastricker** [comment]
I completely agree, nothing to add.

**Comment by ceriottm** [comment]
thanks @rosecers updated tutorials are great. Let's go over this together tomorrow or next week. I think this is also a good point to make a decision on the universal vs per-specie model, as these tutorials set our standard for "this is what we expect a noob to do" thing. 
TBD also with @felixmusil and @max-veit

**Comment by Luthaf** [comment]
I fixed the merge conflicts, but I had to get rid of 347a67b7005d3f9911349953338d3dd7fa7bb3c5 since the corresponding files (`feature_manager`) seems to be gone on master. I can add the functions back if someone points me to the right file to do so!

**Comment by max-veit** [comment]
> I fixed the merge conflicts, but I had to get rid of [347a67b](https://github.com/cosmo-epfl/librascal/commit/347a67b7005d3f9911349953338d3dd7fa7bb3c5) since the corresponding files (`feature_manager`) seems to be gone on master. I can add the functions back if someone points me to the right file to do so!

All of that functionality has been superseded by the Kernel class that was merged in `feat/representation_calculator`, so no worries.

As for the other changes, anything that's not strictly related to documentation -- or merging and getting the code to compile -- should be left to a separate maintenance branch (see e.g. `feat/spellcheck`).

**Comment by max-veit** [comment]
One more thing: Since the progress bar and time/memory estimates seem like they're going to take a significant amount of extra work, should we spin those off into a separate branch (`feat/tutorials`, perhaps) and merge in this one (which was initially intended as merging the Sphinx and Doxygen doc systems)?

**Comment by Luthaf** [comment]
> To this end, the Doxygen HTML docs have been left as a reference, but we eventually want to replace them with the sphinx/breathe equivalents.

Is the HTML still in the PR? This would explain the +77000/-1000 diff 😄  Do you want to remove it before merging?

**Comment by max-veit** [comment]
Oh, yeah, that. I didn’t think the built files got included, though (in
fact, I did remove one set of built files already; are there more?).

But what I think we want to do before merging is to disable the Doxygen
html build by default, just leaving the html file, and add an advanced
build option to turn it on.

Guillaume Fraux <notifications@github.com> schrieb am Do. 19. Sep. 2019 um
18:10:

**Comment by max-veit** [comment]
Wait, no, the actual built documentation isn't included in the project, though there are some HTML template files and some images for the static tutorial that got included. There's also a PDF that we could probably strip out. And don't forget the (sometimes huge) ipynb diffs.

I'm ambivalent about keeping the images; on the one hand, they're useful for showing the result of running the cells in the tutorial without forcing someone to re-run it on their own. But on the other hand, they're big(ish) binary files that contribute to repository bloat. Any thoughts?

File: docs/source/tutorials/tutorials.rst [hunk 1]
```diff
@@ -3,10 +3,24 @@
...
 
 
+Getting started
+~~~~~~~~~~~~~~~
+
+.. toctree::
+    :maxdepth: 1
+
+    TutorialEn
+    SOAP_CSD500.ipynb
```
Comment (agoscinski) [inline docs/source/tutorials/tutorials.rst]: What do you think of putting the explanations of `TutorialEn` also in an ipynb? I would use doxygen tutorials only for explanation referring code inside the python library code as we do it for c++

Comment (max-veit) [reply docs/source/tutorials/tutorials.rst]: Not sure I understand the second sentence. We can rework the tutorials as needed in the new `feat/tutorials` branch.

Comment (max-veit) [reply docs/source/tutorials/tutorials.rst]: And don't forget that static results/images that you'd want to be displayed in the web interface are currently at odds with out ipython notebook stripout policy. Though I think you can get `nbsphinx` to automatically run all notebook cells before converting them to html, even if I'm not sure how much of a pain it is to make sure it's using the correct python environment...

**Comment by max-veit** [comment]
@agoscinski you just committed a huge amount of ipython notebook output; can you remove it please? (clear outputs before committing, at least until the ipynb_hook is merged in)

File: examples/SOAP_example.ipynb [hunk 2]
```diff
@@ -52,271 +20,136 @@
...
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
-    "code_folding": []
-   },
    "outputs": [],
    "source": [
-    "# Load the small molecules \n",
-    "frames = read('./data/small_molecules-1000.xyz',':600')"
+    "mySOAP.plot_prediction()"
```
[the remark this answers is not in the document]
Comment (max-veit) [reply examples/SOAP_example.ipynb]: It's not yet merged with master; there's a bunch of stuff that's bound not to work right away. I'm leaving this to be worked on in `feat/tutorials`.

**Comment by max-veit** [comment]
Okay, pretty much done here. Just need to fix merge conflicts, then rebase to remove the large files. Incidentally, this means I'll need to force-push the branch, so please don't do anything else here until I've finished.

**Comment by max-veit** [comment]
Stripped the large pdf and xyz files (`CSD-500.xyz` is now automatically downloaded when you run the tutorial; perhaps we should turn this off for the time being?). IPython notebooks can now be included as tutorials, and will be automatically run [as long as their outputs are cleared](https://nbsphinx.readthedocs.io/en/0.4.2/executing-notebooks.html). Now rebased and ready for merge -- sorry for the force-push, but it was necessary anyway to strip out the large files.

Further contribution to the documentation is more than welcome, as there's certainly a lot of work to be done. For one, organizing the C++ Sphinx/breathe documentation is an utter pain (though maybe [Exhale](https://exhale.readthedocs.io/en/latest/overview.html) can help us with this?), and formatting is still an issue. Let's make a new branch (say, `feat/documentation_improvements`) for these incremental improvements and updates to the documentation itself. And @Luthaf perhaps we can get the online build with e.g. gh-pages going soon?

Needless to say, if you're working on a file that has either insufficient docstrings or docstrings that don't display properly in the online docs, please do fix them as part of the coding process!

**Pull request merged**